modules/markup/markdown/goldmark.go

```diff
@@ -178,6 +179,11 @@ func (g *ASTTransformer) Transform(node *ast.Document, reader text.Reader, pc pa
 					v.SetHardLineBreak(setting.Markdown.EnableHardLineBreakInDocuments)
 				}
 			}
+		case *ast.CodeSpan:
+			colorContent := n.Text(reader.Source())
+			if css.ColorHandler(strings.ToLower(string(colorContent))) {
+				v.AppendChild(v, NewColorPreview(colorContent))
+			}
```

Comment:
Won't these lines severly impact performance when you have messages with long code spans?
Also, doesn't this limit the usage to explicit `` `rgb(…)` `` or `` `#abcdef` ``?
I thought this was supposed to handle all occurrences of `#abcdef` or `rgb(104,0,30)`?
EDIT: Ah, okay, GitHub handles it the same way

Reply:
Under the hood, it's 4 regex match operations

Reply:
The four regexes are less my problem, more the `ToLower`.
The regexes will terminate once the input doesn't match anymore, so after at most something like 20 chars.
The `ToLower` converts the **whole** code span, no matter how much text it contains.
Especially when you have to render multiple things containing large code spans, this will influence performance.
Additionally, aren't fenced code blocks like this
````
```csv
text
```
````
also code spans? Those can get really long, for example when you paste a diff.
Are the regexes perhaps case-insensitive already?

Reply:
The regex work only for lowercase, that's the reason for `ToLower`. As I understand it, fenced code blocks are **not** code spans. Would you prefer going back to the initial general regex and only if it matches, run the `ToLower`+`bluemonday/css.ColorHandler`?

Reply:
Nope, current approach should be fine.